lux/executor/PandasExecutor.py

```diff
@@ -443,16 +436,16 @@ def compute_data_type(self, ldf: LuxDataFrame):
                     ldf._data_type[attr] = "geographical"
                 elif pd.api.types.is_float_dtype(ldf.dtypes[attr]):
 
-                    if ldf.cardinality[attr] != len(ldf) and (ldf.cardinality[attr] < 20):
+                    if ldf.cardinality[attr] != ldf._length and (ldf.cardinality[attr] < 20):
```

Comment:
What is the difference between `_length` and `len(df)`? It is probably more general to use the latter since the `_length` might not be maintained correctly.

Reply:
I noticed `_length` in the metadata for a LuxDataFrame, and found that it was not being used anywhere in the code base (as far as I could tell). On Line 544 of this file, I changed it to be the length of the sampled DataFrame. This is necessary since we don't save the sampled DataFrame after the metadata is computed, but the length of the sampled DataFrame is necessary for future calculations, especially ones related to cardinality, like the one here.

The name of the attribute is probably not the best, so I could maybe change it to `_sampled_length` instead?